plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,666 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package slurm
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Slurm struct {
+	URL              string          `toml:"url"`
+	Username         string          `toml:"username"`
+	Token            string          `toml:"token"`
+	IgnoredEndpoints []string        `toml:"ignored_endpoints"`
+	ResponseTimeout  config.Duration `toml:"response_timeout"`
+	tls.ClientConfig
+
+	client      *goslurm.APIClient
+	baseURL     *url.URL
+	endpointMap map[string]bool
+}
+
+func (s *Slurm) createHTTPClient(host string) *goslurm.APIClient {
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = host
+	configuration.Scheme = "http"
+	configuration.UserAgent = "Telegraf Metrics Agent"
+	configuration.HTTPClient = &http.Client{
+		Timeout: time.Duration(s.ResponseTimeout),
+	}
+
+	return goslurm.NewAPIClient(configuration)
+}
+
+func (s *Slurm) createHTTPSClient(host string) (*goslurm.APIClient, error) {
+	tlsCfg, err := s.ClientConfig.TLSConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	configuration := goslurm.NewConfiguration()
+	configuration.Host = host
+	configuration.Scheme = "https"
+	configuration.UserAgent = "Telegraf Metrics Agent"
+	configuration.HTTPClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+		},
+		Timeout: time.Duration(s.ResponseTimeout),
+	}
+
+	return goslurm.NewAPIClient(configuration), nil
+}
+
+func (*Slurm) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Slurm) Init() error {
+	if s.ResponseTimeout < config.Duration(time.Second) {
+		s.ResponseTimeout = config.Duration(time.Second * 5)
+	}
+
+	s.endpointMap = map[string]bool{}
+	for _, endpoint := range s.IgnoredEndpoints {
+		s.endpointMap[strings.ToLower(endpoint)] = true
+	}
+
+	if s.URL == "" {
+		return errors.New("empty URL provided")
+	}
+
+	u, err := url.Parse(s.URL)
+	if err != nil {
+		return err
+	}
+
+	if u.Hostname() == "" {
+		return fmt.Errorf("invalid hostname %q", u.Hostname())
+	}
+
+	s.baseURL = u
+
+	switch u.Scheme {
+	case "http":
+		s.client = s.createHTTPClient(u.Host)
+	case "https":
+		s.client, err = s.createHTTPSClient(u.Host)
+		if err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("invalid scheme %q", u.Scheme)
+	}
+
+	return nil
+}
+
+func (s *Slurm) GatherDiagMetrics(acc telegraf.Accumulator,
+	diag *goslurm.V0038DiagStatistics) {
```

Comment:
These methods don't need to be exported
```suggestion
func (s *Slurm) gatherDiagMetrics(acc telegraf.Accumulator,
	diag *goslurm.V0038DiagStatistics) {
```

Reply:
Thanks a ton for pointing that out. If I'm honest I thought about whether it was necessary at some point!

Anyway, these have been unexpected on fe0836991a6fab23decafa72a1f3c25e4941c013.